STX.REST.RESTFulSense.Clients/Models/Services/HttpExchanges/Exceptions/InvalidHttpExchangeRequestException.cs

```diff
@@ -0,0 +1,18 @@
+// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using Xeptions;
+
+namespace STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges.Exceptions
+{
+    public class InvalidHttpExchangeRequestException : Xeption
```

Comment:
This exception model appears unused, remove if not needed.

Reply:
For me this "InvalidHttpExchangeRequestException" should be used instead "InvalidArgumentHttpExchangeException".

Any thought?

I don´t like too much "ValidateInputArgument", I think it should be ValidateHttpExchangeRequest, as i suggested in line before, if we use InvalidHttpExchangeRequestException stays more consistent.

Reply:
Then it seems this comment would be moot, and set to resolved. Let the argument continue. @aosunlana @cjdutoit see @LBoullosa thoughts regarding "InvalidInput".

Reply:
![image](https://github.com/user-attachments/assets/66ca12a9-6c8f-439b-a457-ee305c043bb8)

This is a bit confusing too, because we are not validating a HttpRequestHeader, we are validating a HttpExchangeRequestHeader. I know the naming used is because still exists a method with the same name, but then we should take a think and make it more clear. This is the same as I argumented before for the InvalidArgument and ValidateInputArgument

Reply:
I would not use `InvalidArgumentHttpException` as you are not validating primitive arguments / parameters on the method.  We have a contract here, so we should be using `InvalidHttpExchangeException` for the top level exception.

When we drill down into the validation we can get more specific exceptions thrown for the child objects i.e.

Validate for httpExchange.Request =>  throws `InvalidHttpExchangeRequestException`
Validate for HttpExchange.Request.Headers => throws `InvalidHttpExchangeRequestHeaders`

We do not have input arguments, so we should not have this validation method `ValidateInputArgument`